### PR TITLE
Saulparedes/lsg release

### PR DESCRIPTION
**Pull request by Redent0r**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

###### Change Log  

- Change
- Change
- Change

###### Does this affect the toolchain?  

**YES/NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

File: SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+%global clh_name cloud-hypervisor
+%global clh_type cvm
+
+Name:           %{clh_name}-%{clh_type}
+Summary:        Cloud Hypervisor CVM is an open source Virtual Machine Monitor (VMM) that enables running SEV SNP enabled VMs on top of MSHV using the IGVM file format as payload.
+Version:        %{gitver}.%{gitnum}
+Release:        1000.%{gitrev}%{?dist}
+License:        ASL 2.0 or BSD-3-clause
+Group:          Applications/System
+URL:            https://github.com/cloud-hypervisor/cloud-hypervisor
```
Comment (ms-mahuber) [inline SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec:25]: are we sure we don't want to reference the microsoft fork link?

File: SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+Source0:        https://github.com/microsoft/cloud-hypervisor/archive/refs/tags/msft/v%{version}.tar.gz#/%{clh_name}-%{version}.tar.gz
```
Comment (ms-mahuber) [inline SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec:26]: Our sources may collide with proper cloud-hypervisor. Any chance we can still rename the tar (and vendor) file to contain clh_type?

File: SPECS/cloud-hypervisor-cvm/update_src [hunk 3]
```diff
@@ -0,0 +1,72 @@
+#/bin/bash
```
Comment (ms-mahuber) [inline SPECS/cloud-hypervisor-cvm/update_src:1]: is this file required?

File: SPECS/cloud-hypervisor-cvm/fetch-private-deps [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
```
Comment (ms-mahuber) [inline SPECS/cloud-hypervisor-cvm/fetch-private-deps:1]: is this file required?

File: SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec [hunk 5]
```diff
@@ -0,0 +1,229 @@
+# This spec file assumes you're building on an environment where:
```
Comment (ms-mahuber) [inline SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec:1]: We should remove those comments and in generally align with cloud-hypervisor where possible

File: SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+# * You have rustup installed on your system
+# * You have both x86_64-unknown-linux-gnu and x86_64-unknown-linux-musl, or
+# * You have both aarch64-unknown-linux-gnu and aarch64-unknown-linux-musl
+#   targets installed.
+
+%define using_rustup 0
+%define using_musl_libc 0
+%define using_vendored_crates 1
+
+# Generated by update script
```
Comment (ms-mahuber) [inline SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec:12]: We should remove this comment. Why do we need gitver, gitnum, gitrev variables? Those are only used once below.

File: SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+%global gitver   38.0
+%global gitnum   72
+%global gitrev   8f7209bdc
+%global clh_name cloud-hypervisor
```
Comment (ms-mahuber) [inline SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec:16]: We can get rid of those two variables if we can rename our tar files to avoid ambiguity with the proper CLH tar files

File: SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+%if 0%{?using_vendored_crates}
+%define cargo_offline --offline
+%endif
+
+%description
+Cloud Hypervisor is an open source Virtual Machine Monitor (VMM) that runs on top of KVM. The project focuses on exclusively running modern, cloud workloads, on top of a limited set of hardware architectures and platforms. Cloud workloads refers to those that are usually run by customers inside a cloud provider. For our purposes this means modern Linux* distributions with most I/O handled by paravirtualised devices (i.e. virtio), no requirement for legacy devices and recent CPUs and KVM.
+
+%prep
+
+%autosetup -p1 -n %{clh_name}-%{version}
```
Comment (ms-mahuber) [inline SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec:79]: in the regular CLH package we use %setup. Best to align. In general we can align this block a bit more with its regular CLH counterpart.

File: SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec [hunk 9]
```diff
@@ -0,0 +1,229 @@
...
+pushd %{_builddir}/%{clh_name}-%{version}
+%if 0%{?using_vendored_crates}
+tar xf %{SOURCE1}
+mkdir -p .cargo
+cp %{SOURCE2} .cargo/config.toml
+%endif
+popd
+
+%define ch cloud-hypervisor
```
Comment (ms-mahuber) [inline SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec:88]: Is this %define really required?

File: SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec [hunk 10]
```diff
@@ -0,0 +1,229 @@
...
+
+%install
+rm -rf %{buildroot}
+install -d %{buildroot}%{_bindir}
+install -D -m755  ./target/%{rust_def_target}/release/%{ch} %{buildroot}%{_bindir}
+
+%if 0%{?using_musl_libc}
```
Comment (ms-mahuber) [inline SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec:95]: vhost_user_net, vhost_user_blk is not required here?

File: SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec [hunk 11]
```diff
@@ -0,0 +1,229 @@
...
+cp %{SOURCE2} .cargo/config.toml
+%endif
+popd
+
+%define ch cloud-hypervisor
+
+%install
+rm -rf %{buildroot}
+install -d %{buildroot}%{_bindir}
+install -D -m755  ./target/%{rust_def_target}/release/%{ch} %{buildroot}%{_bindir}
```
Comment (ms-mahuber) [inline SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec:93]: ch_remote, vhost_user_blk, vhost_user_net is not required here?

File: SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec [hunk 12]
```diff
@@ -0,0 +1,229 @@
...
+export OPENSSL_NO_VENDOR=1
+%endif
+cargo build --release --target=%{rust_def_target} %{cargo_pkg_feature_opts} %{cargo_offline}
+
+%if 0%{?using_musl_libc}
+cargo build --release --target=%{rust_musl_target} %{cargo_pkg_feature_opts} %{cargo_offline}
+
+%endif
+
+%clean
```
Comment (ms-mahuber) [inline SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec:142]: if we need this, why does the regular CH SPEC not need this?

**Comment by Redent0r** [comment]
Closing in favor of https://github.com/microsoft/azurelinux/pull/9261

File: SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec (hunk 2 above)
Comment (Redent0r) [reply SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec:26]: They should not because clh-cvm always has the patch number in its version (eg 38.0.72). Vs upstream only uses major and minor (eg 38.0). Still good to make it clear these are cvm sources

File: SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec (hunk 10 above)
Comment (Redent0r) [reply SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec:95]: Looks like we haven't been installing those https://dev.azure.com/mariner-org/mariner/_git/CBL-Mariner-Proprietary?path=/SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec&version=GBmain&line=89&lineEnd=99&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

I'll look into it more

File: SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec (hunk 2 above)
Comment (Redent0r) [reply SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec:26]: Will also need to adapt Kata CI to rename the tars (add `-cvm` ) given by LSG

File: SPECS/cloud-hypervisor-cvm/fetch-private-deps (hunk 4 above)
Comment (Redent0r) [reply SPECS/cloud-hypervisor-cvm/fetch-private-deps:1]: Double checked and doesn't look like we need this. This seems to be a helper script used by LSG to fetch the sources from the package feed

File: SPECS/cloud-hypervisor-cvm/update_src (hunk 3 above)
Comment (Redent0r) [reply SPECS/cloud-hypervisor-cvm/update_src:1]: Double checked and no. Same as https://github.com/microsoft/azurelinux/pull/9161/files#discussion_r1621506207. We can get the source tar by looking at the tags page in https://github.com/microsoft/cloud-hypervisor/tags , and we can generate the vendor tar by following https://github.com/microsoft/azurelinux/pull/9261/files#diff-5d949bbc496656a772fe7a22601f9837efcc1dffcf513a7ccffd6aa6d53ae2e4R16-R22 (adding this comment to cvm spec)

May be worth documenting this comment in the form of a useful script for the future. It's 3 or 4 straightforward commands

File: SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec (hunk 7 above)
Comment (Redent0r) [reply SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec:16]: Would like to take this a step forward and remove gitrev from release number altogether. We can find that info by looking at the version used in https://github.com/microsoft/cloud-hypervisor/tags eg

image

This would also further align with normal clh

File: SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec (hunk 2 above)
Comment (Redent0r) [reply SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec:26]: Just double checked and this isn't as straightforward as expected. The macro `%setup` expects the untared folder to also be called *-cvm

image

And LSG doesn't renames them to *-cvm. We could rename them ourselves, but this would alter the hash sum. This makes me want to leave them as is (eg `cloud-hypervisor-38.0.72.tar.gz`), and rely on the fact that there won't be a name collision with normal clh sources, because clh-cvm will always have a patch number in its source tar

File: SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec (hunk 11 above)
Comment (Redent0r) [reply SPECS/cloud-hypervisor-cvm/cloud-hypervisor-cvm.spec:93]: ch-remote: utility to interact with cloud-hypervisor api (https://www.cloudhypervisor.org/docs/prologue/commands/#ch-remote-binary). Seems we can already do that using curl https://dev.azure.com/mariner-org/mariner/_wiki/wikis/mariner.wiki/2539/Booting-the-UVM-manually-on-a-node-via-CH-API-socket

vhost_user_blk and vhost_user net: They seem to be "backends" that provide functionality for network and block devices (https://www.qemu.org/docs/master/system/devices/vhost-user.html) . I think kata(cc) handles network and block devices differently and doesn't need these. Also considering they weren't being installed before
